File: app/src/main/java/com/lhs94/pictonote/NoteListAdapter.java
package com.lhs94.pictonote;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.lhs94.pictonote.note.Note;
import com.lhs94.pictonote.note.NoteActivity;

import java.util.ArrayList;

public class NoteListAdapter extends RecyclerView.Adapter {

    private ArrayList<Note> notes;

    private Context context;

    NoteListAdapter(Context context) {
        this.context = context;
        notes = new ArrayList<>();
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.item_note_list, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        ViewHolder vh = (ViewHolder)holder;
        Note note = notes.get(position);
        vh.position = position;
        vh.title.setText(note.getTitle());
        vh.text.setText(note.getText());
        Uri uri = note.getThumbnailUri();

        Glide.with(context).load(uri).error(R.drawable.image_empty).into(vh.thumbnail);
    }

    @Override
    public int getItemCount() {
        return notes.size();
    }

    void updateData(ArrayList<Note> notes) {
        this.notes = notes;
        notifyDataSetChanged();
    }

    private class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        int position;

        ImageView thumbnail;
        TextView title;
        TextView text;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            itemView.setOnClickListener(this);

            thumbnail = itemView.findViewById(R.id.thumbnail);
            title  =  itemView.findViewById(R.id.title);
            text = itemView.findViewById(R.id.text);
        }

        @Override
        public void onClick(View v) {
            Intent intent = new Intent(context, NoteActivity.class);
            intent.putExtra("note", notes.get(position));
            context.startActivity(intent);
        }
    }
}

File: app/src/main/java/com/lhs94/pictonote/ui/main/NoteListAdapter.kt
package com.lhs94.pictonote.ui.main

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.databinding.DataBindingUtil
import androidx.navigation.findNavController
import androidx.recyclerview.widget.RecyclerView
import com.lhs94.pictonote.R
import com.lhs94.pictonote.databinding.ItemNoteListBinding
import com.lhs94.pictonote.room.entity.Note
import java.util.*

class NoteListAdapter : RecyclerView.Adapter<NoteListViewHolder>() {
    var notes: ArrayList<Note> = arrayListOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteListViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val v = DataBindingUtil.inflate<ItemNoteListBinding>(layoutInflater, R.layout.item_note_list, parent, false)
        v.adapter = this
        return NoteListViewHolder(v)
    }

    override fun onBindViewHolder(holder: NoteListViewHolder, position: Int) {
        holder.binding.apply {
            note = notes[position]
            index = position
        }
        //val uri = note.thumbnailUri
        //Glide.with(vh.thumbnail.context).load(uri).error(R.drawable.image_empty).into(vh.thumbnail)
    }

    override fun getItemCount(): Int {
        return notes.size
    }

    fun updateData(notes: List<Note>) {
        this.notes.clear()
        this.notes.addAll(notes)
        notifyDataSetChanged()
    }

    fun onClickNoteListItem(item: View, index: Int) {
        val note = notes[index]
        item.findNavController().navigate(R.id.noteFragment, bundleOf("note" to note))
    }
}

class NoteListViewHolder(val binding: ItemNoteListBinding) : RecyclerView.ViewHolder(binding.root)

